### PR TITLE
#GSSoC2023 1.Added section 'ml,ai and data' and provided links for ML, Data Science and Deep Learning 

### PR DESCRIPTION
**Pull request by Grogu22**
…, dl and data science'

2. also added a section for DeepLearningAI in ```database/youtube/machine-learning.json```

3. fixed a bug in displaying the ```golang``` section in main branch

4. added a new blog card for ```blogs.json```

5. modified the ```types/index.ts```, ```data.ts``` and ```index.ts``` sections for displaying the changes

## Changes proposed
Issue : #614 
1. Created a new category **ML, AI & DATA**
2. Created subcategories for _Machine Learning_, _Deep Learning_ and _Data Science_
3. Created a new entry in the `blogs.json` file for _Andrew Ng's blog_ on AI, ML and Deep Learning
4. Created a new entry in `machine-learning.json` file for DeepLearningAI's youtube channel
5. Added a link in `database/languages/golang.json` file, which was empty [🙃](https://emojipedia.org/upside-down-face/)

## Screenshots

image

image

image

image

image
image
image

## Note to reviewers

Please comment if some issues persist.. will fix them ASAP !!

### CODE REVIEW

**Comment by Grogu22** [comment]
@rupali-codes @CBID2  , created a _ML, Deep Learning, Data Science_ related category and added a few other stuff !!
Thought it was important ..

**Comment by CBID2** [comment]
> @rupali-codes @CBID2  , created a _ML, Deep Learning, Data Science_ related category and added a few other stuff !!
> 
> Thought it was important ..

Good job @Grogu22! 😊

**Comment by Grogu22** [comment]
> > @rupali-codes @CBID2  , created a _ML, Deep Learning, Data Science_ related category and added a few other stuff !!
> > Thought it was important ..
> 
> Good job @Grogu22! 😊

also, can I get write access to the repo please.. [😢](https://emojipedia.org/crying-face/)
cannot add a GSSoC'23 label !!

File: database/youtube/machine-learning.json [hunk 1]
```diff
@@ -39,6 +39,14 @@
         "category":"youtube",
         "subcategory":"machine-learning",
         "language":"english"
+    },
+    {
+        "name":"DeepLearningAI",
+        "description":"Access the lectures of Andrew Ng on Deep Learning and Machine Learning",
```
Comment (CBID2) [inline database/youtube/machine-learning.json:45]: [Typo]
```suggestion
        "description": "Access the lectures of Andrew Ng on Deep Learning and Machine Learning",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hey @Grogu22! 😊 Aside from the one typo I mentioned in my inline comment, your PR is off to a great start! 😊

**Comment by CBID2** [comment]
> > > @rupali-codes @CBID2  , created a _ML, Deep Learning, Data Science_ related category and added a few other stuff !!
> 
> > > Thought it was important ..
> 
> > 
> 
> > Good job @Grogu22! 😊
> 
> 
> 
> also, can I get write access to the repo please.. [😢](https://emojipedia.org/crying-face/)
> 
> cannot add a GSSoC'23 label !!

Sorry @Grogu22. Only the maintainers can have write access to the repo.

**Comment by Grogu22** [comment]
yup.. committed the changes @CBID2

**Review by CBID2 — APPROVED** [review]
Looks great @Grogu22! 😄 LGTM! 😄

**Comment by CBID2** [comment]
> yup.. committed the changes @CBID2

Link the issue that your PR pertains to as well

**Comment by Grogu22** [comment]
Did not open an issue.. did a direct PR!! Should I open one now? @CBID2

**Comment by CBID2** [comment]
> Did not open an issue.. did a direct PR!! Should I open one now? @CBID2

Yes @Grogu22. I'm going to close this PR for now. Once you create the issue, I'll open it right back up! 👍🏾 In the future, please thoroughly review the [Contributing.md](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md) before you contribute to LinksHub again.

**Comment by Grogu22** [comment]
This PR solves the issue #614.
Maybe merge it @CBID2 @rupali-codes ?

File: database/data.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { v4 as uuidv4 } from 'uuid'
 export const database: IData[][] = Object.values(DB).map((item: IDBData[]) =>
   item.map((subcat) => ({ ...subcat, id: uuidv4() }))
 )
-
+console.log(database)
```
Comment (rupali-codes) [inline database/data.ts:8]:
```suggestion
```

File: database/data.ts [hunk 3]
```diff
@@ -35,6 +35,14 @@ export const sidebarData: ISidebar[] = [
       { name: 'system design', url: '/system-design' },
     ],
   },
+  {
+    category: 'ml, ai & data',
+    subcategory: [
+      { name: 'ml', url: '/ml' },
```
Comment (rupali-codes) [inline database/data.ts:41]:
```suggestion
      { name: 'machine learning', url: '/ml' },
```

File: database/data.ts [hunk 4]
```diff
@@ -35,6 +35,14 @@ export const sidebarData: ISidebar[] = [
...
+      { name: 'data-science', url: '/data-science'},
```
Comment (rupali-codes) [inline database/data.ts:42]:
```suggestion
      { name: 'data-science', url: '/data-science'},
```
```suggestion
      { name: 'data science', url: '/data-science'},
```

File: database/data.ts [hunk 5]
```diff
@@ -35,6 +35,14 @@ export const sidebarData: ISidebar[] = [
...
+      { name: 'deep-learning', url : '/deep-learning'},
```
Comment (rupali-codes) [inline database/data.ts:43]:
```suggestion
      { name: 'deep learning', url : '/deep-learning'},
```

**Comment by Grogu22** [comment]
Hey, @CBID2 @rupali-codes, did the following:
1. Added issue to PR
2. Resolved conversations
3. Resolved a merge conflict

**Review by CBID2 — APPROVED** [review]
LGTM! 😊

**Review by rupali-codes — APPROVED** [review]
looks good

**Pull request merged**